src/modules/jcms_rest/tests/src/SubjectsRestResourceTest.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+namespace Drupal\Tests\jcms_rest\Unit;
+
+use Drupal\Tests\UnitTestCase;
+use Drupal\jcms_rest\Plugin\rest\resource\SubjectsRestResource;
+use \Mockery as m;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class SubjectsRestResourceTest
+ *
+ * @package Drupal\Tests\jcms_rest\Unit
+ */
+class SubjectsRestResourceTest extends UnitTestCase {
+
+  /**
+   * @var \Drupal\jcms_rest\Plugin\rest\resource\SubjectsRestResource
+   *   The class we're testing.
+   */
+  public $resource;
+
+  protected $queryFactory;
+
+  protected $query;
+
+  protected $requestStack;
+
+  protected $entityManager;
+
+  protected $entityTypeManager;
+
+  public function setUp() {
+    //$this->term = m::mock('Drupal\taxonomy\Entity\Term');
+    // Set the container.
+    $container = new ContainerBuilder();
+    // Entity field query.
+    $this->queryFactory = m::mock('Drupal\Core\Entity\Query\QueryFactory');
+    // Returned query (normally mock Drupal\Core\Entity\Query\Sql\Query but this
+    // causes the test to error).
```

Comment:
I guess this is the "lots of mocking" you were referring to. It seems it is isolating the resource class code from the database.
My evaluation also is that this kind of test provides a low value/cost ratio, consider also that many of the mocked method calls could change without the behavior actually being incorrect (e.g. using similar method find() instead of a get(), as a result of refactoring).
So my suggestion is to take a look at the duplicated code between resources, extract that into classes (someone said composition over inheritance yesterday?) and unit test those; resource code, which to me is similar to controllers in MVC frameworks, can probably be best tested at the API level.